lib/imagefs.coffee

```diff
@@ -23,16 +23,71 @@ _ = require('lodash')
 filedisk = require('file-disk')
 fs = Promise.promisifyAll(require('fs'))
 replaceStream = require('replacestream')
+ext2fs = require('ext2fs')
+
 driver = require('./driver')
+utils = require('./utils')
 
 checkImageType = (image) ->
 	if not (_.isString(image) or image instanceof filedisk.Disk)
 		throw new Error('image must be a String (file path) or a filedisk.Disk instance')
 
+composeDisposers = (outerDisposer, createInnerDisposer) ->
+	Promise.resolve(outerDisposer)
+	.then (outerDisposer) ->
+		outerDisposer._promise
+		.then (outerResult) ->
+			Promise.resolve(createInnerDisposer(outerResult))
+			.then (innerDisposer) ->
+				innerDisposer._promise
+				.then (innerResult) ->
+					Promise.resolve(innerResult)
+					.disposer (innerResult) ->
+						Promise.resolve(innerDisposer._data(innerResult))
+						.then ->
+							outerDisposer._data(outerResult)
+			.catch (err) ->
+				outerDisposer._data(outerResult)
+				throw err
```

Comment:
Why call `outerDisposer._data(outerResult)` again here, without handling anything, and then still throw?